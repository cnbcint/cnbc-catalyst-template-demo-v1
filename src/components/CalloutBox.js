import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin } from "../helpers/Layout";
const Container = styled.div`
  flex: 1 1 auto;
  background-color: ${(props) => props.backgroundColor};
  padding: ${getMargin(1)};
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
`;
const Pic = styled.img`
  border-radius: 50%;
  width: 25% !important;
  margin: 0 !important;
`;

const Person = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  align-self: flex-start;
  margin-top: auto;
`;
const Details = styled.div`
  width: 75%;
  box-sizing: border-box;
  > * {
    display: block;
  }
`;
const Spacer = styled.div`
  width: 25% !important;
`;
const Copy = styled.div`
  margin-bottom: ${getMargin(3)};
`;
export default function CalloutBox(props) {
  return (
    <Container backgroundColor={props.backgroundColor}>
      <Copy>{props.copy()}</Copy>
      <Person>
        <Details>
          <strong>{props.fullname}</strong>
          <small>{props.position}</small>
          <small>{props.company}</small>
        </Details>
        {props.profilePicURL && (
          <Pic src={props.profilePicURL} alt="proile pic" />
        )}
        {!props.profilePicURL && <Spacer />}
      </Person>
    </Container>
  );
}

CalloutBox.defaultProps = {
  backgroundColor: "#ccc",
};

CalloutBox.propTypes = {
  copy: PropTypes.func,
  backgroundColor: PropTypes.string,
  fullname: PropTypes.string,
  position: PropTypes.string,
  company: PropTypes.string,
  profilePicURL: PropTypes.string,
};
