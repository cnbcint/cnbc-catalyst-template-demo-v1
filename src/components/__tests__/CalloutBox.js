/* eslint react/display-name: 0 */
import React from "react";
import CalloutBox from "../CalloutBox";
import { render } from "@testing-library/react";
import "jest-styled-components";

const testProps = {
  fullname: "Jimmy Fallon",
  profilePicURL: "http://localhost:8080/images/jimmy_fallon.jpg",
  position: "Position",
  company: "NBC Universal",
  copy: () => (
    <>
      “Lorem ipsum dolor sit amet, aster si consectetur adipiscing elit. Donec
      malesuada blandit diam vel pharetra. Maecenas in bibendum sapien, eu di
      iaculis ligula. Duis tempor diam ex, non fringilla nibh faucibus et”
    </>
  ),
};
describe("CalloutBox", () => {
  it("Renders correctly", () => {
    const { container } = render(<CalloutBox {...testProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
