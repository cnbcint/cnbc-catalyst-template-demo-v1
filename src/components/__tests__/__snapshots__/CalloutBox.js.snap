// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalloutBox Renders correctly 1`] = `
.c0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  background-color: #ccc;
  padding: 1rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  border-radius: 50%;
  width: 25% !important;
  margin: 0 !important;
}

.c2 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  margin-top: auto;
}

.c3 {
  width: 75%;
  box-sizing: border-box;
}

.c3 > * {
  display: block;
}

.c1 {
  margin-bottom: 3rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    “Lorem ipsum dolor sit amet, aster si consectetur adipiscing elit. Donec malesuada blandit diam vel pharetra. Maecenas in bibendum sapien, eu di iaculis ligula. Duis tempor diam ex, non fringilla nibh faucibus et”
  </div>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <strong>
        Jimmy Fallon
      </strong>
      <small>
        Position
      </small>
      <small>
        NBC Universal
      </small>
    </div>
    <img
      alt="proile pic"
      class="c4"
      src="http://localhost:8080/images/jimmy_fallon.jpg"
    />
  </div>
</div>
`;
